.github/workflows/build.yml

```diff
@@ -39,33 +39,48 @@ jobs:
+          # TODO: for some reason windows has a different path for the generators folder than linux
+          if [ "${{ env.OPERATING_SYSTEM }}" == "linux" ]; then
+            if [ "${{ env.COMPILER }}" == "msvc" ]; then
+              cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G "Visual Studio 17 2022" -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-msvc -DCMAKE_C_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-msvc -DCMAKE_TOOLCHAIN_FILE=build/$BUILD_TYPE/generators/conan_toolchain.cmake
+            elif [ "${{ env.COMPILER }}" == "clang" ]; then
+              cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_CC=clang -DCMAKE_CXX=clang++ -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-gnu -DCMAKE_C_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-gnu -DCMAKE_TOOLCHAIN_FILE=build/$BUILD_TYPE/generators/conan_toolchain.cmake
+            elif [ "${{ env.COMPILER }}" == "gcc" ]; then
+              cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-gnu -DCMAKE_C_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-gnu -DCMAKE_TOOLCHAIN_FILE=build/$BUILD_TYPE/generators/conan_toolchain.cmake
+            fi
+          elif [ "${{ env.OPERATING_SYSTEM }}" == "windows" ]; then
+            if [ "${{ env.COMPILER }}" == "msvc" ]; then
+              cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G "Visual Studio 17 2022" -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-msvc -DCMAKE_C_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-msvc -DCMAKE_TOOLCHAIN_FILE=build/generators/conan_toolchain.cmake
+            elif [ "${{ env.COMPILER }}" == "clang-cl" ]; then
+              cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_MODULE_PATH=build/generators/ -DCMAKE_PREFIX_PATH=build/generators/
+            elif [ "${{ env.COMPILER }}" == "gcc" ]; then
+              cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_CXX_COMPILER=g++ -DCMAKE_MODULE_PATH=build/generators/ -DCMAKE_PREFIX_PATH=build/generators/
+            fi
           fi
 
       - name: Build
         run: cmake --build build --parallel --config ${{ env.BUILD_TYPE }}
 
       - name: Test
-        # TODO: figure out why unit testing doesn't work on windows clang
-        if: |
-            !(env.OPERATING_SYSTEM == 'windows' && env.COMPILER == 'clang')
         run: ctest --test-dir build --build-config ${{ env.BUILD_TYPE }} --output-on-failure
 
       - name: Regression
-        # TODO: figure out why regression testing doesn't work on windows gnu or windows clang
-        if: |
-            !(env.OPERATING_SYSTEM == 'windows' && (env.COMPILER == 'gcc' || env.COMPILER == 'clang'))
```

Comment:
MinGW GCC will fail to find its C++ runtime DLL when running any novatel_edie executables unless it is explicitly handled. I've done so here: https://github.com/novatel/novatel_edie/blob/d15f0b7bd714c5d3868fac390448be95fbbbb18c/CMakeLists.txt#L125-L139

Reply:
thanks!